tasks/service-runner/app/service/service-runner.ts

```diff
@@ -0,0 +1,201 @@
+import * as k8s from '@kubernetes/client-node';
+import { readdirSync } from 'fs';
+import stream from 'stream';
+import env from '../util/env';
+import logger from '../../../../app/util/log';
+import WorkItem from '../../../../app/models/work-item';
+
+// Must match where harmony expects artifacts in workflow-orchestration.ts
+const ARTIFACT_DIRECTORY = '/tmp/metadata';
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+
+const exec = new k8s.Exec(kc);
+
+export interface ServiceResponse {
+  batchCatalogs?: string[];
+  error?: string;
+}
+
+// how long to let a worker run before giving up
+const { workerTimeout } = env;
+
+class LogStream extends stream.Writable {
+  logStr = '';
+
+  shouldLog = true;
+
+  _write(chunk, enc: BufferEncoding, next: (error?: Error | null) => void): void {
+    const chunkStr = chunk.toString('utf8');
+    this.logStr += chunkStr;
+    if (this.shouldLog) {
+      logger.debug(`FROM WORKER LOG: ${chunkStr}`);
+    }
+    next();
+  }
+}
+
+/**
+ * Get a list of STAC catalog in a directory
+ * @param dir - the directory containing the catalogs
+ */
+function _getStacCatalogs(dir: string): string[] {
+  // readdirSync should be ok since a service only ever handles one WorkItem at a time and may
+  // actually be necessary to ensure read after write consistency on EFS
+  return readdirSync(dir)
+    .filter((fileName) => fileName.match(/catalog\d*.json/))
+    .map((fileName) => `${dir}/${fileName}`);
+}
+
+/**
+ * Parse an error message out of an error log
+ *
+ * @param logStr - A string that contains error logging
+ * @returns An error message parsed from the log
+ */
+function _getErrorMessage(logStr: string): string {
+  // expect JSON logs entries
+  const regex = /\{"message":.*?"application":.*?\}/g;
+  const matches = logStr.match(regex);
+  for (const match of matches) {
+    const logEntry = JSON.parse(match);
+    if (logEntry.level === 'ERROR') {
+      return logEntry.message;
+    }
+  }
+
+  return 'Unknown error';
+}
+
+/**
+ * Run the query cmr service for a work item pulled from Harmony
+  * @param operation - The requested operation
+  * @param callback - Function to call with result
+  */
+export function runQueryCmrFromPull(workItem: WorkItem): Promise<ServiceResponse> {
```

Comment:
This one-off bugs me a little, but not so much as to reject this merge.  Should we ever need a third version of a "run the work" method, let's step back and figure out a better way

Reply:
ok